### PR TITLE
mgmt, container service upgrade to 2023-03-01

### PR DESCRIPTION
**Pull request by haolingdong-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by haolingdong-msft** [comment]
revapi error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2714363&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=68a849d6-f425-5f38-d450-7e2c3d1df885&l=11

**Comment by haolingdong-msft** [comment]
revapi error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2714363&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=ba3cf43a-b2bf-50f9-f95f-93049ca44ad1&l=846

**Comment by haolingdong-msft** [comment]
> > revapi error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2714363&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=ba3cf43a-b2bf-50f9-f95f-93049ca44ad1&l=846
> 
> capture the important ones in changelog. what are the 2 that get removed?
I will capture those breaking changes into changelog. 

The two get removed are:
field com.azure.resourcemanager.containerservice.models.ManagedClusterSkuName.BASIC
field com.azure.resourcemanager.containerservice.models.ManagedClusterSkuTier.PAID

I checked spec repo, the two enum values are removed in 2023-03.
2023-03:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-03-01/managedClusters.json#L4940
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-03-01/managedClusters.json#L4920

2023-01:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-01-01/managedClusters.json#L5124
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-01-01/managedClusters.json#L5148

**Comment by weidongxu-microsoft** [comment]
I think that get approved. Just log it in changelog. There are actually real breaks (customer can no longer set them)

File: sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@
 
 ### Breaking Changes
 
+- Removed field `BASIC` from class `ManagedClusterSkuName`.
+- Removed field `PAID from class `ManagedClusterSkuTier`.
+- Changed class `ManagedClusterAddonProfileIdentity` to be `final`.
+- Changed class `ManagedClusterIdentity` to be `final`.
+- Changed class `ManagedClusterPodIdentity` to be `final`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md:13]: these actually don't need to be mentioned.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@
 
 ### Breaking Changes
 
+- Removed field `BASIC` from class `ManagedClusterSkuName`.
+- Removed field `PAID from class `ManagedClusterSkuTier`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md:10]: check again!

Comment (haolingdong-msft) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/CHANGELOG.md]: updated, thanks

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**